src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11623,7 +11632,8 @@ void Assembler::vex_prefix(Address adr, int nds_enc, int xreg_enc, VexSimdPrefix
   }
 }
 
-int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexSimdPrefix pre, VexOpcode opc, InstructionAttr *attributes) {
+int Assembler::vex_prefix_and_encode(int dst_enc, int nds_enc, int src_enc, VexSimdPrefix pre, VexOpcode opc, InstructionAttr *attributes, bool has_gpr, bool src_is_gpr) {
+  bool is_extended = has_gpr && (dst_enc >= 16 || nds_enc >= 16 || src_enc >=16);
```

Comment:
This could be:
     bool is_extended = dst_enc >= 16 || nds_enc >= 16 || src_enc >=16;

Reply:
The line below at 11647 could be changed from:
if ((!attributes->uses_vl() || (attributes->get_vector_len() != AVX_512bit)) &&
          (dst_enc < 16) && (nds_enc < 16) && (src_enc < 16)) {
to
if ((!attributes->uses_vl() || (attributes->get_vector_len() != AVX_512bit)) &&
          !is_extended) {